src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs

```diff
@@ -77,20 +77,34 @@ public override void VisitUsingStatement(UsingStatementIRNode node)
 
         public override void VisitCSharpStatement(CSharpStatementIRNode node)
         {
-            if (node.Source != null)
+            if (node.Children.Count == 0)
             {
-                using (Context.Writer.BuildLinePragma(node.Source.Value))
+                return;
+            }
+
+            for (var i = 0; i < node.Children.Count; i++)
+            {
+                if (!(node.Children[i] is RazorIRToken token) ||
+                    !token.IsCSharp)
                 {
-                    var padding = BuildOffsetPadding(0, node.Source.Value, Context);
-                    Context.Writer.Write(padding);
+                    continue;
+                }
 
-                    Context.AddLineMappingFor(node);
-                    Context.Writer.Write(node.Content);
+                if (token.Source != null)
+                {
+                    using (Context.Writer.BuildLinePragma(token.Source.Value))
+                    {
+                        var padding = BuildOffsetPadding(0, token.Source.Value, Context);
+                        Context.Writer.Write(padding);
+
+                        Context.AddLineMappingFor(token);
+                        Context.Writer.Write(token.Content);
+                    }
+                }
+                else
+                {
+                    Context.Writer.WriteLine(token.Content);
```

Comment:
Tokens should not get their own line. We want to be able to generate statements that contain a mix of tokens that have a line mapping and are synthesized.